mlflow/sklearn/__init__.py

```diff
@@ -500,3 +506,118 @@ def should_log(self):
         return (self._parent is None) or (
             self._parent.allow_children and self._parent.clazz != self.clazz
         )
+
+
+@experimental
+def autolog():
```

Comment:
@dbczumar This sounds like a follow-up task (because this PR is already gigantic), but we could emit a warning when a user has already imported estimators before enabling autolog.

```python
import warnings

from sklearn.linear_model import LinearRegression  # import estimator before enabling autolog

# mlflow.sklearn.autolog()

def is_estimator_class(obj):
    from sklearn.base import BaseEstimator

    return isinstance(obj, type) and (obj != BaseEstimator) and issubclass(obj, BaseEstimator)


estimators_imported = list(filter(is_estimator_class, globals().values()))

if len(estimators_imported):
    warnings.warn("< warning message to encourage users to enable autolog before importing estimators >")
```